riff-raff/app/controllers/api.scala

```diff
@@ -347,4 +348,52 @@ class Api(deployments: Deployments)(implicit val messagesApi: MessagesApi, val w
     )
   }
 
+  def validate = ApiJsonEndpoint("validate") { implicit request =>
+    request.body.asText.fold{
+      Json.obj(
+        "response" -> Json.obj(
+          "status" -> "error",
+          "errors" -> Json.arr("No configuration provided in request body")
+        )
+      )
+    }{ body =>
+      val validatedGraph = YamlResolver.resolveDeploymentGraph(body)
+      validatedGraph match {
+        case Valid(graph) =>
+          Json.obj(
+            "response" -> Json.obj(
+              "status" -> "ok",
+              "result" -> "passed",
+              "deployments" -> graph.toList.map { deployment =>
+                Json.obj(
+                  "name" -> deployment.name,
+                  "type" -> deployment.`type`,
+                  "stacks" -> deployment.stacks.toList,
+                  "regions" -> deployment.regions.toList,
+                  "actions" -> deployment.actions,
+                  "app" -> deployment.app,
+                  "contentDirectory" -> deployment.contentDirectory,
+                  "dependencies" -> deployment.dependencies,
+                  "parameters" -> deployment.parameters
+                )
+              }
+            )
+          )
+        case Invalid(errors) =>
+          Json.obj(
+            "response" -> Json.obj(
+              "status" -> "ok",
```

Comment:
Yes. I *think* so. It has successfully given you a validation result. That result might be a failure, but it is still a 200 response which in the API world is OK. See the case above where you don't even give it valid data to validate in which case you do get an error.